src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/DateTimeMarshaler.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript
+{
+    public class DateTimeMarshaler
+    {
+        public static string FromJavaScriptPreFilter { get; } = "value.toISOString()";
+        public static string ToJavaScriptPostFilter { get; } = "new Date(value)";
+
+        public static DateTime FromJavaScript (string s)
+        {
+            return DateTime.Parse(s).ToUniversalTime();
+        }
+
+        public static string ToJavaScript (ref DateTime dt)
```

Comment:
I think the current marshallers were samples not really meant to be the final form but I think they are worth discussing because trying to do them well runs into other problems.  For many reasons I think something like:

```suggestion
        public static double ToJavaScript (in DateTime dt) => (dt.Ticks - DateTime.UnixEpoch.Ticks)/10000; 
```

is preferable here. It should be much faster and should marshal as the correct date even if the settings between js and managed are different.  

and going the other way would look something like
```csharp
public static double FromJavaScriptPreFilter { get; } = "value.valueOf() * 10000";
```

Reply:
I'm fine with micro-optimizing this if we need to, but is ISO 8601 date string parsing actually culture dependent? It shouldn't be, there's a whole spec.

The previous code was optimized in this way, but it was also broken. When I tried to write automated tests to prove that it worked (before starting on the marshaler API), it turned out that our DateTime marshalers didn't actually work correctly.

Reply:
I was coming at this from the perspective of correctness not only optimization.  I agree the previous marshallers didn't work which is why I don't care about breaking the already broken case.

Reply:
After looking at it more, I think ISO strings are still the correct way to do this. We can switch to numbers if we have a performance reason to do it, IMO. The behavior is well-specified and obvious vs doing int64 math and double conversions.